hudi-common/src/main/java/org/apache/hudi/common/fs/HoodieSerializableFileStatus.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.fs;
+
+import org.apache.hadoop.fs.FileStatus;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+public class HoodieSerializableFileStatus extends FileStatus implements Serializable {
+  FileStatus status;
```

Comment:
Can we add some doc for it, can we reuse this clazz for Spark ?

Reply:
It might be unnecessary to reuse this class for Spark part becasue: 1) That code is borrowed from [Spark](https://github.com/apache/spark/blob/5103e00c4ce5fcc4264ca9c4df12295d42557af6/core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala#L156) 2) The Spark `SerializableFileStatus` is tailored for `LocatedFileStatus` instead of vanilla `FileStatus` so it also uses `SerializableBlockLocation`. I think reusing it there means we'll need to have `HoodieSerializableFileStatus` to hold more states but ideally we want to keep it as simple as possible so it can just be a serializable wrapper of `FileStatus`.

I'll add doc to `HoodieSerializableFileStatus`